ledger/src/blockstore_db.rs

```diff
@@ -303,13 +302,29 @@ impl Rocks {
                         db_options.set_max_open_files(-1);
 
                         Rocks(
-                            DB::open_cf_as_secondary(&db_options, path, &secondary_path, names)?,
+                            DB::open_cf_as_secondary(
+                                &db_options,
+                                path,
+                                &secondary_path,
+                                cf_names.clone(),
+                            )?,
                             ActualAccessType::Secondary,
                         )
                     }
                 }
             }
         };
+        for cf_name in cf_names {
+            db.0.set_options_cf(
+                db.cf_handle(cf_name),
+                &[(
+                    "ttl",
```

Comment:
https://github.com/facebook/rocksdb/wiki/Leveled-Compaction#ttl

some promising quotes

> A dynamic ttl column-family option has been introduced to solve this problem. Files (and, in turn, data) older than TTL will be scheduled for compaction when there is no other background work. This will make the data go through the regular compaction process and get rid of old unwanted data. This also has the (good) side-effect of all the data in the non-bottommost level being newer than ttl, and all data in the bottommost level older than ttl. Note that it could lead to more writes as RocksDB would schedule more compactions.

and some un-promisng one:

> A file could exist in the LSM tree without going through the compaction process for a really long time if there are no updates to the data in the file's key range.

at least, we're issuing deleterange for the older slot range.... this is a bit of contradiction. but I think vastly large amount writes near higher slot numbers (rocksdb's keys) could skew this selection algorithm.

Reply:
And more commentary from rocksdb:

https://github.com/facebook/rocksdb/commit/77eab5c85aa7a05d2200339738373f8b0b921e81:

> Make default value of options.ttl to be 30 days

Previously, this was set as 30-days by default by rocksdb. So, no wonder we saw increasing ledger dir in the order of weeks (`--limit-ledger-size`'s targeted data retention window)


https://github.com/facebook/rocksdb/commit/4c9aa30a62d004ad0addb83dba062941783fc201

> But it is occasionally possible that the compactions get delayed by considerable time due to factors like low writes to a key range, data reaching bottom level, etc before the TTL expiry.

So, this is my backing theory for no compaction in our case too. it indicates like small write (only deleterange in our case) might be not enough for compaction triggering.

Reply:
also confusing enough, this `ttl` is completely different thing from this `ttl`: https://github.com/facebook/rocksdb/wiki/Time-to-Live

Reply:
> https://github.com/facebook/rocksdb/wiki/Leveled-Compaction#ttl
> 
> some promising quotes
> 
> > A dynamic ttl column-family option has been introduced to solve this problem. Files (and, in turn, data) older than TTL will be scheduled for compaction when there is no other background work. This will make the data go through the regular compaction process and get rid of old unwanted data. This also has the (good) side-effect of all the data in the non-bottommost level being newer than ttl, and all data in the bottommost level older than ttl. Note that it could lead to more writes as RocksDB would schedule more compactions.

@steviez, fyi, the first section of the page in this link is a good overview of how compaction in RocksDb works